libdevcore/CommonData.h

```diff
@@ -50,14 +50,34 @@ template <class T, class U> std::vector<T>& operator+=(std::vector<T>& _a, U&& _
 	std::move(_b.begin(), _b.end(), std::back_inserter(_a));
 	return _a;
 }
+/// Concatenate the contents of a container onto a multiset
+template <class U, class ...T> std::multiset<T...>& operator+=(std::multiset<T...>& _a, U const& _b)
+{
+	_a.insert(_b.begin(), _b.end());
+	return _a;
+}
+/// Concatenate the contents of a container onto a multiset, move variant.
+template <class U, class ...T> std::multiset<T...>& operator+=(std::multiset<T...>& _a, U&& _b)
+{
+	for (auto&& x: _b)
+		_a.insert(std::move(x));
```

Comment:
If we use `std::forward` here, do we actually need the above variant?

Reply:
I think so because even with `std::forward` we can't use this one with `const` objects (if I am not mistaken?)

Reply:
``&&`` to ``const&`` is ``const&``, so if called with a const object ``U`` will be deduced as ``const SomeType&`` and ``const SomeType& &&`` is ``const SomeTyp&`` again... aka it will work with ``std::forward`` (you can read up on "perfect forwarding" for details).

Reply:
As we discussed in the chat: since ``_b`` is actually the container and ``x`` will never be deduced as rvalue reference (it'll always either be a const or a non-const lvalue reference), forwarding here will in fact be tricky, so probably not worth it...